File: 프로그래머스/1/12926. 시저 암호/시저 암호.cs
using System;
using System.Text;
using System.Linq;

public class Solution {
    public string solution(string s, int n) {
        
        StringBuilder builder = new StringBuilder();
        
        foreach (char c in s)
        {
            if (char.IsWhiteSpace(c))
            {
                builder.Append(c);
                continue;
            }
            
            char shiftCharacter = (char)(c + n);
            
            if (char.ToLower(c) + n > (int)'z')
            {
                builder.Append((char)(shiftCharacter - 26));
                continue;
            }
            
            builder.Append(shiftCharacter);
        }
        
        return builder.ToString();
        
        // return string.Concat(s.Select(c => char.IsWhiteSpace(c)? c : (char)(char.ToLower(c) + n > (int)'z' ? c + n - 26 : c + n)));
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    
    for (int i = 0; i < s.size(); i++)
    {
        char ch = s[i];

        if (ch == ' ')
        {
            answer.push_back(ch);
            continue;
        }


        if (tolower(ch) + n > 'z')
        {
            answer.push_back(ch + (n % 26) - 26);
            continue;
        }

        answer.push_back(ch + n);
    }
    
    return answer;
}